File: Jin/week04/day16/땅따먹기.java
package Jin.week04.day16;

import java.util.*;

class 땅따먹기 {
    int solution(int[][] land) {
        for (int i = 1; i < land.length; i++) {
            for (int j = 0; j < land[0].length; j++) {
                int max = 0;
                for (int k = 0; k < land[0].length; k++) {
                    if (j == k) continue;
                    max = Math.max(max, land[i - 1][k]);
                }
                land[i][j] += max;
            }
        }

        return Arrays.stream(land[land.length - 1]).max().getAsInt();
    }
}

File: Jiho/week12/day60/땅따먹기.js
function solution(land) {
  let dp = Array.from({ length: land.length }, () =>
    Array.from({ length: 4 }, () => 0)
  );

  // 1. dp 첫째줄 초기화
  dp[0] = [...land[0]];

  // 2. dp 채우기
  for (let i = 1; i < land.length; i++) {
    for (let j = 0; j < 4; j++) {
      dp[i][j] =
        land[i][j] + Math.max(...dp[i - 1].filter((_, index) => index !== j));
    }
  }

  return Math.max(...dp[dp.length - 1]);
}
